src/transformations.jl

```diff
@@ -8,63 +8,84 @@ abstract type AbstractDataTransform end
 
 Apply transformation `t` to vector or matrix `x` in place.
 """
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(x, t, x)
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(x, t, x)
+function transform!(t::AbstractDataTransform, x::AbstractMatrix{<:Real})
+    transform!(x, t, x)
+    return x
+end
+
+function transform!(t::AbstractDataTransform, x::AbstractVector{<:Real})
+    transform!(t, reshape(x, :, 1))
+    return x
+end
 
 """
     transform(t::AbstractDataTransform, x)
 
-Return a row-standardized vector or matrix `x` using `t` transformation.
+Return a standardized vector or matrix `x` using `t` transformation.
 """
-transform(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(similar(x), t, x)
-transform(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(similar(x), t, x)
+transform(t::AbstractDataTransform, x::AbstractMatrix{<:Real}) = transform!(similar(x), t, x)
+
+function transform(t::AbstractDataTransform, x::AbstractVector{<:Real})
```

Comment:
Not going to reshape to matrix which means that I have to implement a 3-argument `transform!` for vector, right? Isn't it just the way I have been done before and got suggestion to change to use `reshape`? I got confused.

Reply:
I think the best approach is to define the 3-argument `transform!` for vectors and have reshape the vector to a matrix and pass it to `transform!` for matrices. *But* the vector function should return the original vector, *not* the reshaped matrix. The reshaping is just an internal trick to avoid code duplication.

Reply:
OK, that's clear.